librosa/core/pitch.py

```diff
@@ -333,3 +337,462 @@ def piptrack(y=None, sr=22050, S=None, n_fft=2048, hop_length=None,
+
+    .. [#] De Cheveigné, Alain, and Hideki Kawahara.
+        "YIN, a fundamental frequency estimator for speech and music."
+        The Journal of the Acoustical Society of America 111.4 (2002): 1917-1930.
+
+    Parameters
+    ----------
+    y : np.ndarray [shape=(n,)]
+        audio time series.
+
+    fmin: number > 0 [scalar]
+        minimum frequency in Hertz.
+        The recomended minimum is ``librosa.note_to_hz('C2')`` (~65 Hz)
+        though lower values may be feasible.
+
+    fmax: number > 0 [scalar]
+        maximum frequency in Hertz.
+        The recomended maximum is ``librosa.note_to_hz('C7')`` (~2093 Hz)
+        though higher values may be feasible.
+
+    sr : number > 0 [scalar]
+        sampling rate of ``y`` in Hertz.
+
+    frame_length : int > 0 [scalar]
+         length of the frames in samples.
+         By default, ``frame_length=2048`` corresponds to a time scale of about 93 ms at
+         a sampling rate of 22050 Hz.
+
+    win_length : None or int > 0 [scalar]
+        length of the window for calculating autocorrelation in samples.
+        If ``None``, defaults to ``frame_length // 2``
+
+    hop_length : None or int > 0 [scalar]
+         number of audio samples between adjacent YIN predictions.
+         If ``None``, defaults to ``frame_length // 4``.
+
+    trough_threshold: number > 0 [scalar]
+        absolute threshold for peak estimation.
+
+    center : boolean
+        If ``True``, the signal `y` is padded so that frame
+        ``D[:, t]`` is centered at `y[t * hop_length]`.
+        If ``False``, then ``D[:, t]`` begins at ``y[t * hop_length]``.
+        Defaults to ``True``,  which simplifies the alignment of ``D`` onto a
+        time grid by means of ``librosa.core.frames_to_samples``.
+
+    pad_mode : string or function
+        If ``center=True``, this argument is passed to ``np.pad`` for padding
+        the edges of the signal ``y``. By default (``pad_mode="reflect"``),
+        ``y`` is padded on both sides with its own reflection, mirrored around
+        its first and last sample respectively.
+        If ``center=False``,  this argument is ignored.
+        .. see also:: `np.pad`
+
+    Returns
+    -------
+    f0: np.ndarray [shape=(n_frames,)]
+        time series of fundamental frequencies in Hertz.
+
+    See Also
+    --------
+    librosa.pyin
+        Fundamental frequency (F0) estimation using probabilistic YIN (pYIN).
+
+    Examples
+    --------
+    Computing a fundamental frequency (F0) curve from an audio input
+
+    >>> y = librosa.chirp(440, 880, duration=5.0)
+    >>> librosa.yin(y, 440, 880)
+    array([442.66354675, 441.95299983, 441.58010963, ...,
+        871.161732  , 873.99001454, 877.04297681])
+    '''
+    # Set the default window length if it is not already specified.
+    if win_length is None:
+        win_length = frame_length // 2
+
+    if win_length > frame_length:
+        raise ParameterError('win_length cannot exceed frame_length')
```

Comment:
It'd be great to have the provided values here:

`'win_length={} cannot exceed given frame_length={}'.format(win_length, frame_length)`